### DISCUSSION TITLE
Yourls Front End

### DISCUSSION THREAD

**Discussion by rkingg-dev** [YOURLS/YOURLS#3455] [General]
Hello! First I want to thank all the contributor and the author of this awesome YOURLS.

But how can I change the front end of YOURLS? Cant find the HTML Structure so I can edit it on my end. Tried using Sleeky but still I want to change the appearance of it. 

TIA

**Comment by dgw** [comment]
Best approach is to inspect the HTML in your browser's developer tools. That will also let you apply tweaks and see the results live, without needing to edit a file and refresh the page.

I believe YOURLS v2 (in development) is planned to have "more official" support for theming (@LeoColomb correct me if this is wrong, please!)

**Comment by ozh** [comment]
Not sure what you call "front end". YOURLS ships with an example "front end", to show you how you would do your own. See [sample-public-front-page.txt](https://github.com/YOURLS/YOURLS/blob/master/sample-public-front-page.txt) and code your own.

The admin part, what I would personally call the "back end", is generated with functions found in `includes/functions-html.php`.

Sleeky has both custom front and back end, probably the best way to see how it's done is to dissect and butcher their files.

**Comment by rkingg-dev** [comment]
Got it! Thank you!
